src/batch-client/index.ts

```diff
@@ -169,6 +169,49 @@ function normalizeMessage(
 	);
 }
 
+const hasUnreadDescendent = (folder: any) => {
+	if (
+		folder[
+			folder.name === 'Drafts' || folder.name === 'Outbox'
+				? 'nonFolderItemCount'
+				: 'unread'
+		] > 0
+	) {
+		return true;
+	}
+
+	const folderArray = get(folder, 'folders') || [];
+	for (let i = 0; i < folderArray.length; i++) {
```

Comment:
one small optimization to not count length with every iteration
`for (let i = 0, len = folderArray.length; i < len; i++) {`

Reply:
Done